docs/dev_guide/migrate-a-project/user-defined-migration-rules.rst

```diff
@@ -215,4 +232,32 @@ output source code:
    int *ptr, *ptr2;
    int * new_ptr = bar(*ptr, 30);
 
+.. _pattern_rewriter_rule_ref:
 
+Post-migration Pattern-Rewrite
```

Comment:
This section doesnt really add much new information. 

The example rule is identical to the rule defined above (post_migration_rewriter_rule) - why not just update the rule above with all the comments from below (which is what provides all the new info)? Then this section can be removed. 
(No other rule type gets this kind of special treatment). If more info is needed re type, it could be added in the Key/Value table. Glad to discuss this.

Reply:
Pattern-rewriter is different from other rules.
This kind of rule will be applied to the migrated SYCL code while other rules are applied to the original CUDA code.
Therefore, I tried to make the rule looks differents in the doc to make sure users are aware of the difference.

Reply:
I have removed the session in the new commit.